core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -154,6 +155,31 @@ static HttpResponse delayed(HttpResponse response, Duration delay, ScheduledExec
         return res;
     }
 
+    /***
+     * Invokes the specified {@link Supplier} and creates a new HTTP response that
+     * delegates to the provided {@link HttpResponse} by {@link Supplier},
+     * beginning publishing after {@code delay} has passed from a random {@link ScheduledExecutorService}.
+     */
+    static HttpResponse delayed(Supplier<? extends HttpResponse> responseSupplier, Duration delay) {
+        requireNonNull(responseSupplier, "responseSupplier");
+        requireNonNull(delay, "delay");
+        return delayed(responseSupplier.get(), delay, CommonPools.workerGroup().next());
+    }
+
+    /***
+     * Invokes the specified {@link Supplier} and creates a new HTTP response that
+     * delegates to the provided {@link HttpResponse} {@link Supplier},
+     * beginning publishing after {@code delay} has passed from the provided {@link ScheduledExecutorService}.
+     */
+    static HttpResponse delayed(Supplier<? extends HttpResponse> responseSupplier,
+                                Duration delay,
+                                ScheduledExecutorService executor) {
+        requireNonNull(responseSupplier, "responseSupplier");
+        requireNonNull(delay, "delay");
+        requireNonNull(executor, "executor");
+        return delayed(responseSupplier.get(), delay, executor);
```

Comment:
We need to lazily invoke `responseSupplier.get()` when the `delay` comes like:
```java
executor.schedule(() -> responseSupplier.get().delegate(response), delay.toNanos(), TimeUnit.NANOSECONDS);
```

Reply:
I applied it, thank you 😄